src/frontend/src/pages/PIADetailPage/index.tsx

```diff
@@ -71,13 +92,67 @@ const PIADetailPage = () => {
         }
       }
     })();
-  }, [id, navigate, pia.hasAddedPiToDataElements, pia.ministry, win]);
+  }, [
+    id,
+    navigate,
+    pia.hasAddedPiToDataElements,
+    pia.ministry,
+    pia.status,
+    win,
+  ]);
 
   const handleAlertClose = () => {
     setFetchPiaError('');
     setDownloadError('');
   };
 
+  const handleEdit = () => {
+    // the status will change to enum when Brandon pr merged
+    if (
+      isMPO &&
+      piaStatus !== 'INCOMPLETE' &&
+      piaStatus !== 'EDIT_IN_PROGRESS'
+    ) {
+      setModalConfirmLabel(messages.Modal.ConfirmLabel.en);
+      setModalCancelLabel(messages.Modal.CancelLabel.en);
+      setModalTitleText(messages.Modal.TitleText.en);
+      setModalParagraph(messages.Modal.ParagraphText.en);
+      setShowModal(true);
+    } else {
+      navigate(`${routes.PIA_INTAKE}/${id}`, {
```

Comment:
as we discussed, can we please create a `${routes.PIA_INTAKE}/${id}/edit` route?